core/src/main/scala/tech/beshu/ror/configuration/EsConfigFileLoader.scala

```diff
@@ -20,16 +20,25 @@ import better.files.File
 import cats.Show
 import cats.data.NonEmptyList
 import cats.implicits._
-import io.circe.Decoder
+import io.circe.{Decoder, Json}
 import tech.beshu.ror.accesscontrol.factory.JsonConfigStaticVariableResolver
 import tech.beshu.ror.accesscontrol.factory.JsonConfigStaticVariableResolver.ResolvingError
 import tech.beshu.ror.providers.EnvVarsProvider
 import tech.beshu.ror.utils.yaml
 
-final class EsConfigFileLoader[CONFIG: Decoder]()(implicit envVarsProvider: EnvVarsProvider) {
+final class EsConfigFileLoader(file: File)
+                              (implicit envVarsProvider: EnvVarsProvider) {
 
-  def loadConfigFromFile(file: File,
-                         configName: String): Either[MalformedSettings, CONFIG] = {
+  def loadConfig[CONFIG: Decoder](configName: String): Either[MalformedSettings, CONFIG] = {
+    loadEsConfigJson
+      .flatMap { json =>
+        implicitly[Decoder[CONFIG]]
+          .decodeJson(json)
+          .left.map(_ => MalformedSettings(s"Invalid $configName configuration"))
```

Comment:
OK, it looks like I understand it now. We are loading yaml file and then trying to decode it to `CONFIG`. 
I'd ask you about some more improvements here:
1. Let's change MalformedSettings's message to: "Cannot load $configName from file ${file.pathAsString}" (and make sure that all `configName`s make sense in this message context
2. let's rename EsConfigFileLoader to YamlFileBasedConfigLoader (and remove all "es" from names)
3. cache result of `loadEsConfigJson` (lazy val?)